crates/forge_infra/src/executor.rs

```diff
@@ -1,26 +1,50 @@
-use std::io::{self, Write};
+use std::io;
 use std::path::{Path, PathBuf};
 use std::sync::Arc;
 
 use forge_domain::{CommandOutput, Environment};
 use forge_services::CommandInfra;
-use tokio::io::AsyncReadExt;
 use tokio::process::Command;
 use tokio::sync::Mutex;
 
+use crate::stream_service::{StreamService, stream_to_writer};
+
 /// Service for executing shell commands
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub struct ForgeCommandExecutorService {
     restricted: bool,
     env: Environment,
+    stdout_stream_service: Option<Arc<dyn StreamService>>,
+    stderr_stream_service: Option<Arc<dyn StreamService>>,
 
```

Comment:
Create a new `execute_streaming` method that returns a stream as the commands are executing.